cmd/uniq/uniq_test.go

```diff
@@ -0,0 +1,315 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"testing"
+)
+
+var input = []byte(`hello
+world
+
+hello
+
+世界
+世界
+世
+1
+3
+4
+日本語
+4
+1
+`)
+
+func newLine(s string, ln []int) *Line {
+	newStr := new(string)
```

Comment:
string is immutable and `new` is deprecated.
This function could be created something like:

```go
func newLine(s string, ln []int) Line {
        line := s+"\n"
        return Line{
                text: &line,
                nums: ln,
        }
}
```
In Go every variable in analysed if it scapes the function and if so it's allocated in the heap and managed by GC. You don't need to call `new` anymore (since go1.3 or earlier).